.github/workflows/sort-locale-json.yaml

```diff
@@ -0,0 +1,50 @@
+name: Sort JSON on Pull Requests
+
+on:
+  pull_request:
+    paths:
+      - "client/packages/common/src/intl/locales/**/*.json" # Trigger only for locale json changes
+
+permissions:
+  contents: write
+
+jobs:
+  sort-json:
+    name: Sort Json
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Echo branch
+        run: |
+          echo ${{ github.ref }}
+          echo ${{ github.head_ref }}
+
+      - name: Sort translation files
+      - uses: sergeysova/jq-action@v2
+        with:
+          cmd: |
+            TRANSLATION_FILES=($(find ./client/packages/common/src/intl/locales/ -type f -name "*.json"))
+            for file in "${TRANSLATION_FILES[@]}"; do
+              echo "Sorting $file..."
+              jq --sort-keys . "$file" > "${file}.tmp" && mv "${file}.tmp" "$file"
+            done
+
+      - name: Commit and push changes
+        run: |
+          git fetch origin
+          git checkout ${{ github.head_ref }} || git checkout -b ${{ github.head_ref }} origin/${{ github.head_ref }}
+          # Check for changes
+          if [ "$(git status --porcelain)" ]; then
+            echo "Changes detected, committing..."
+            git config --global user.name "github-actions[bot]"
+            git config --global user.email "github-actions[bot]@users.noreply.github.com"
+            git add .
+            git commit -m "chore: sorted JSON files"
+            git push https://x-access-token:${{ secrets.PAT }}@github.com/${{ github.repository }} HEAD:${{ github.head_ref }}
+          else
+            echo 'No changes to commit.'
+          fi
```

Comment:
an alternative to this block is simply:

```
      - uses: stefanzweifel/git-auto-commit-action@v5
        with:
          commit_message: 'Sorted JSON translation files'
          commit_options: '--no-verify'
```

when I was playing around I did a `git add $file` in the loop - you can try and see if that's needed or no